fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -712,10 +710,15 @@ struct EncryptedRangeFileWriter : public IRangeFileWriter {
 		return std::make_pair(FDB_DEFAULT_ENCRYPT_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME);
 	}
 
-	static Future<std::pair<int64_t, TenantName>> getEncryptionDomainDetails(
-	    KeyRef key,
-	    Reference<TenantEntryCache<Void>> tenantCache) {
-		return getEncryptionDomainDetailsImpl(key, tenantCache);
+	static Future<std::pair<int64_t, TenantName>> getEncryptionDomainDetails(KeyRef key,
+	                                                                         EncryptedRangeFileWriter* self) {
+		// If tenants are disabled on a cluster then don't use the TenantEntryCache as it will result in alot of
```

Comment:
> If tenants are disabled on a cluster then don't use the TenantEntryCache as it will result in alot of

 I think the better way to phrase this would be: `Optimization: for clusters not supporting any tenants, skip tenantCache lookup.
 
 > It is strongly recommended NOT to use snapshot backups with TenantMode::OPTIONAL_TENANT for now as it can also result in many cache misses

I don't think above statement is correct, for `optional` tenant mode, if most of the mutations do belong to a valid tenant, then the first lookup would miss the cache, however, all following lookup would lead to cache-hit. I think the better way to rephrase this could be 
`For a cluster configured in TenantMode::Optional, the backup performance may degrade if most of the mutations belong to an `invalid` tenant`

Reply:
Comment update seems to be missing I think?